### PR TITLE
Skeletons in datumaro format

### PR DESCRIPTION
**Pull request by Eldies**
### Motivation and context

Fixes https://github.com/cvat-ai/cvat/issues/8115

### Checklist

- [ ] I submit my changes into the `develop` branch
- [ ] I have created a changelog fragment 
- [ ] I have updated the documentation accordingly
- [ ] I have added tests to cover my changes
- [ ] I have linked related issues (see [GitHub docs](
  https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword))
- [ ] I have increased versions of npm packages if it is necessary
  ([cvat-canvas](https://github.com/cvat-ai/cvat/tree/develop/cvat-canvas#versioning),
  [cvat-core](https://github.com/cvat-ai/cvat/tree/develop/cvat-core#versioning),
  [cvat-data](https://github.com/cvat-ai/cvat/tree/develop/cvat-data#versioning) and
  [cvat-ui](https://github.com/cvat-ai/cvat/tree/develop/cvat-ui#versioning))

### License

- [ ] I submit _my code changes_ under the same [MIT License](
  https://github.com/cvat-ai/cvat/blob/develop/LICENSE) that covers the project.
  Feel free to contact the maintainers if that's a concern.

## Summary by CodeRabbit

- **New Features**
  - Added support for exporting and importing Skeletons in the Datumaro format.
  
- **Bug Fixes**
  - Improved testing for dataset export and import to cover multiple formats using parameterization.

- **Documentation**
  - Updated documentation to reflect the new "Skeletons" annotation type in Datumaro format.

- **Chores**
  - Updated multiple dependencies to their latest versions for improved security and performance.

- **Tests**
  - Enhanced test coverage for dataset export and import functionality with new formats.

### CODE REVIEW

File: tests/python/rest_api/test_projects.py [hunk 1]
```diff
@@ -568,44 +568,29 @@ def test_can_import_dataset_in_org(self, admin_user):
...
-        }
-
-        self._test_import_project(admin_user, project_id, "CVAT 1.1", import_data)
-
-    def test_can_export_and_import_dataset_with_skeletons_cvat_for_video(self, admin_user):
+    @pytest.mark.parametrize(
+        "export_format, import_format",
+        (
+            ("COCO Keypoints 1.0", "COCO Keypoints 1.0"),
+            ("CVAT for images 1.1", "CVAT 1.1"),
```
Comment (Eldies) [inline tests/python/rest_api/test_projects.py:575]: I have some doubts here: previously the test for CVAT for images relied on default format. Should I keep this behaviour?

File: cvat/requirements/base.txt [hunk 2]
```diff
@@ -1,11 +1,13 @@
...
+# SHA1:e320604118ebbbb6f9a0460d977e416d4cb27200
 #
 # This file is autogenerated by pip-compile-multi
 # To update, run:
 #
 #    pip-compile-multi
 #
 -r ../../utils/dataset_manifest/requirements.txt
+--no-binary lxml
+--no-binary xmlsec
```
Comment (zhiltsov-max) [inline cvat/requirements/base.txt:10]: Please remove these changes in requirements, except `production`. This is handled manually for now, it's a compatibility problem in the requirements update command we use currently.

File: tests/python/rest_api/test_projects.py (hunk 1 above)
Comment (zhiltsov-max) [reply tests/python/rest_api/test_projects.py:575]: It's ok, the CVAT formats are supposed to be default. I think it's better now. One thing here that can be improved is that the source annotations can be checked to contain skeletons. It's either through the `annotations` fixture, or the explicit request to `/api/(projects|tasks|jobs)/{id}/annotations`.

File: cvat/requirements/base.txt (hunk 2 above)
Comment (Eldies) [reply cvat/requirements/base.txt]: Removed

File: tests/python/rest_api/test_projects.py (hunk 1 above)
Comment (Eldies) [reply tests/python/rest_api/test_projects.py:575]: Now the test searches for project with skeleton annotations

File: tests/python/rest_api/test_projects.py [hunk 3]
```diff
@@ -568,44 +568,29 @@ def test_can_import_dataset_in_org(self, admin_user):
...
+        response = self._test_export_project(admin_user, project_id, format=export_format)
 
         tmp_file = io.BytesIO(response.data)
         tmp_file.name = "dataset.zip"
         import_data = {
             "dataset_file": tmp_file,
         }
 
-        self._test_import_project(admin_user, project_id, "CVAT 1.1", import_data)
+        self._test_import_project(admin_user, project_id, import_format, import_data)
```
[the remark this answers is not in the document]
Comment (Eldies) [reply tests/python/rest_api/test_projects.py]: not hardcoded anymore

**Pull request merged**